classes/controller/class-enform-fields-list-table.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Contains Enform_Fields_List_Table class declaration.
+ *
+ * @package P4GBKS\Controllers
+ */
+
+namespace P4GBKS\Controllers;
+
+if ( ! class_exists( '\WP_List_Table' ) ) {
+	require_once ABSPATH . 'wp-admin/includes/class-wp-list-table.php';
+}
+
+use P4GBKS\Controllers\Ensapi_Controller as Ensapi;
+
+/**
+ * Class Enform_Fields_List_Table.
+ * Creates a list table for available en fields overriding WordPress List Table.
+ *
+ * @package P4GBKS\Controllers
+ * @since 1.8.0
+ *
+ * @see \WP_List_Table
+ * @link https://developer.wordpress.org/reference/classes/wp_list_table/
+ */
+class Enform_Fields_List_Table extends \WP_List_Table {
+
+	/**
+	 * Store errors from en api.
+	 *
+	 * @var string $error
+	 */
+	private $error;
+
+	/**
+	 * Enform_Fields_List_Table constructor.
+	 */
+	public function __construct() {
+		parent::__construct(
+			[
+				'ajax' => false,
+			]
+		);
+
+		$this->error = '';
+	}
+
+	/**
+	 * Implements parent's abstract function.
+	 * Prepares the list of items for displaying.
+	 *
+	 * @see \WP_List_Table::prepare_items
+	 */
+	public function prepare_items() {
+		$supporter_fields = [];
+		$main_settings    = get_option( 'p4en_main_settings' );
+
+		if ( isset( $main_settings['p4en_private_api'] ) ) {
+			$ens_private_token = $main_settings['p4en_private_api'];
+			$ens_api           = new Ensapi( $ens_private_token );
+			$supporter_fields  = $ens_api->get_supporter_fields();
+
+			if ( ! is_array( $supporter_fields ) ) {
+				$this->error = $supporter_fields . ' : ' . __( 'Could not fetch results from engaging networks', 'planet4-engagingnetworks-backend' );
```

Comment:
We can change it but it would also require including the EN plugin's translation files into the blocks plugin under the right domain, else text would be untranslated.

Reply:
Changing the text-domain name is doable afterward, So I keep the `planet4-engagingnetworks` and `planet4-engagingnetworks-backend` text-domain as it is in code and copy language  files from EN blocks plugin into blocks plugin.